publish-mqtt/src/main.rs

```diff
@@ -187,77 +191,150 @@ impl Sensor {
                 Self::PROPERTY_ID_TEMPERATURE,
                 format!("{:.2}", readings.temperature),
             )
-            .await?;
+            .await
+            .with_context(|| std::line!().to_string())?;
         homie
             .publish_value(&node_id, Self::PROPERTY_ID_HUMIDITY, readings.humidity)
-            .await?;
+            .await
+            .with_context(|| std::line!().to_string())?;
         homie
             .publish_value(
                 &node_id,
                 Self::PROPERTY_ID_BATTERY,
                 readings.battery_percent,
             )
-            .await?;
+            .await
+            .with_context(|| std::line!().to_string())?;
         Ok(())
     }
 }
 
-async fn bluetooth_mainloop(mut homie: HomieDevice) -> Result<(), Box<dyn Error>> {
-    let sensor_names = hashmap_from_file(SENSOR_NAMES_FILENAME)?;
-
-    let bt_session = BluetoothSession::create_session(None)?;
-    let device_list = scan(&bt_session).await?;
-    let sensors = find_sensors(&bt_session, &device_list);
-    print_sensors(&sensors, &sensor_names);
-    let (named_sensors, unnamed_sensors): (Vec<_>, Vec<_>) = sensors
-        .into_iter()
-        .map(|d| Sensor::new(&d, &sensor_names).unwrap())
-        .partition(|sensor| sensor_names.contains_key(&sensor.mac_address));
-    println!(
-        "{} named sensors, {} unnamed sensors",
-        named_sensors.len(),
-        unnamed_sensors.len()
-    );
-
-    let mut sensors_to_connect: VecDeque<_> = named_sensors.into();
+#[derive(Debug)]
+struct SensorState {
+    sensors_to_connect: VecDeque<Sensor>,
+    sensors_connected: Vec<Sensor>,
+    homie: HomieDevice,
+}
 
-    homie.ready().await?;
+async fn bluetooth_mainloop(
+    mut homie: HomieDevice,
+    dbus_conn: Arc<SyncConnection>,
+) -> Result<(), anyhow::Error> {
+    let sensor_names = hashmap_from_file(SENSOR_NAMES_FILENAME)?;
 
-    let mut sensors_connected: Vec<Sensor> = vec![];
+    homie
+        .ready()
+        .await
+        .with_context(|| std::line!().to_string())?;
+
+    let state = Arc::new(Mutex::new(SensorState {
```

Comment:
- [ ] Think about renaming our structs and enums

we currently have:
* SensorState - contains a Vec and a VecDeque and homie
* Sensor - various props/state about a single sensor
* SensorProps - the mac address and dbus path